Add initial runbooks for all critical CMO alerts (#16)
openshift/runbooks: alerts/cluster-monitoring-operator/PrometheusTargetSyncFailure.md

# PrometheusTargetSyncFailure

## Meaning

This alert is triggered when at least one of the Prometheus instances has
consistently failed to sync its configuration.

## Impact

Metrics and alerts may be missing or inaccurate.

## Diagnosis

Determine whether the alert is for the cluster or user workload Prometheus by
inspecting the alert's `namespace` label. The namespace for the cluster
Prometheus is `openshift-monitoring`, while the namespace for the user workload
monitoring instance is `openshift-user-workload-monitoring`.

Check the logs for the appropriate Prometheus instance:

```console
$ NAMESPACE='<value of namespace label from alert>'

$ oc -n $NAMESPACE logs -l 'app=prometheus'
```

## Mitigation

> If the logs indicate a syntax or other configuration error, correct the

Reviewer: Do we have an example of what that would look like?

Author: Added some sample output.

> # PrometheusTargetSyncFailure

Reviewer 2: this alert is new in 4.9 and it can probably be demoted to warning. For prometheus-k8s, it would only fire if an operator provisioned a buggy service monitor but this should be caught by the CI and there's little that a cluster admin can do to fix the issue.
@RiRa12621 WDYT?

Reviewer: i agree, but the runbook can still stay imho

## Meaning

> This alert is triggered when either the cluster Prometheus or the user workload
> monitoring Prometheus instances have consistently failed to sync their
> configuration.

Reviewer 2: ```suggestion
This alert is triggered when at least one of the Prometheus instances has consistently failed to sync their
configuration.
```

## Impact

Metrics and alerts may be missing or inaccurate.

## Diagnosis

Determine whether the alert is for the cluster or user workload Prometheus by
inspecting the alert's `namespace` label. The namespace for the cluster
Prometheus is `openshift-monitoring`, while the namespace for the user workload
monitoring instance is `openshift-user-workload-monitoring`.

> Check the logs for appropriate Prometheus instance:

Reviewer 2: ```suggestion
Check the logs for the appropriate Prometheus instance:
```
